### PR TITLE
Pal 1041 maven dependencies and testcontainers

### PR DESCRIPTION
**Pull request by dev930018**
* changed maven property to `alpakka.kafka.version` (in line with incoming s3 stuff)
* added testResources maven property filtering
* updated dependency versions to:
``` xml
<akka.version>2.6.10</akka.version>
<alpakka.kafka.version>2.1.0-M1</alpakka.kafka.version>
<testcontainers.version>1.15.3</testcontainers.version>
```
* updated testcontainer image versions to:
``` xml
<testcontainers.kafka.image>confluentinc/cp-kafka:5.5.1</testcontainers.kafka.image>
<testcontainers.redis.image>redis:6-alpine</testcontainers.redis.image>
```
* added testcontainer maven version and default fallbacks to kafka and redis tests
* removed intellij `@NotNull` in favour of spring `@NonNull`

### CODE REVIEW

File: NOTICES.md [hunk 1]
```diff
@@ -38,7 +32,8 @@ List of third-party dependencies grouped by their license type
 
 ### [MIT License](./licenses/mit_license.html):
 * Testcontainers :: Kafka ([org.testcontainers:kafka:1.14.3](https://testcontainers.org))
-* Testcontainers Core ([org.testcontainers:testcontainers:1.14.3](https://testcontainers.org))
+* Testcontainers :: Kafka ([org.testcontainers:kafka:1.15.3](https://testcontainers.org))
+* Testcontainers Core ([org.testcontainers:testcontainers:1.15.3](https://testcontainers.org))
```
Comment (pd104923) [inline NOTICES.md]: I've not been able to identify why there are 2 versions of this but it's not a blocker.

**Review by pd104923 — DISMISSED** [review]
I've approved but there is a comment on the NOTICES file.

File: NOTICES.md (hunk 1 above)
Comment (ac74475) [reply NOTICES.md]: When I run a build the 1.14.3 version gets removed from this file

Comment (dev930018) [reply NOTICES.md]: The kafka specialization (`KafkaContainer`) is distinct from our usage for redis (`GenericContainer`)

**Review by ac74475 — APPROVED** [review]

**Pull request merged**